packages/components/src/components/header/header.scss

```
@@ -53,16 +53,21 @@
 		}
 	}
 
-	// Hide the burger menu if there is no navigation
-	&:has(.db-header-navigation:empty) {
+	// Hide the burger ONLY when the drawer would be empty.
+	// This requires: no main navigation AND no meta navigation AND no secondary action.
+	&:has(.db-header-navigation:empty):has(
+			.db-header-meta-navigation:empty
+		):has(.db-header-secondary-action:empty) {
```

```suggestion
	&:has(.db-header-navigation:empty, .db-header-meta-navigation:empty, .db-header-secondary-action:empty) {
```

You could [concatenate those in a selector-list within one `:has`](https://developer.mozilla.org/en-US/docs/Web/CSS/Selector_list#relative_selector_list).